icu4c/source/data/buildtool/filtration.py

```diff
@@ -126,6 +132,91 @@ def _should_include(self, file_stem):
             return True
 
 
+class UnionFilter(Filter):
+    def __init__(self, json_data):
+        # Collect the sub-filters.
+        self.sub_filters = []
+        for filter_json in json_data["unionOf"]:
+            self.sub_filters.append(Filter.create_from_json(filter_json))
+
+    def match(self, file):
+        """Match iff any of the sub-filters match."""
+        for filter in self.sub_filters:
+            if filter.match(file):
+                return True
+        return False
+
+
+LANGUAGE_SCRIPT_REGEX = re.compile(r"^([a-z]{2,3})_[A-Z][a-z]{3}$")
+LANGUAGE_ONLY_REGEX = re.compile(r"^[a-z]{2,3}$")
+
+class LocaleFilter(Filter):
+    def __init__(self, json_data):
+        self.locales_requested = set()
+        self.locales_required = set()
+        self.include_children = json_data.get("includeChildren", True)
+        self.include_scripts = json_data.get("includeScripts", False)
+
+        # Compute the requested and required locales.
+        for locale in json_data["whitelist"]:
+            self._add_locale_and_parents(locale)
+
+    def _add_locale_and_parents(self, locale):
+        # Store the locale as *requested*
+        self.locales_requested.add(locale)
```

Comment:
I believe that this set is unnecessary because locales_required is a proper superset. You should be able to use just the latter and get the same results. In which case you could (optionally) rename that to just "locales".

Reply:
Discussed.